[박준수] Week2

Jnnsu opened the pull request:
## 요구사항

### 기본

- [x] Github에 1 ~ 2주차 위클리 미션 Pull Request를 만들었나요?
- [x] Netlify에 파일 배포가 아닌 포크한 Github 레포지토리로 연결했나요?
- [x] 아래로 스크롤 해도 “Linkbrary” 로고와 “로그인” 버튼이 있는 상단 네비게이션 바(Global Navigation Bar)영역이 최상단에 고정되게 했나요?
- [x] Linkbrary 로고 클릭시 루트 페이지(“/”)로 이동하나요?
- [x] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일한가요?
- [x] input 요소에 focus in 일 때, 테두리 색상이 파랑색인가요?
- [x] input 요소에 focus out 일 때, 테두리 색상이 회색인가요?
- [x] SNS 아이콘들은 클릭시 각각 “https://www.google.com/”, “https://www.kakaocorp.com/page/” 으로 이동하나요?
- [x] “회원 가입하기”는 클릭시 “/signup” 페이지로 이동하나요?
- [x] “로그인 하기”는 클릭시 “/signin” 페이지로 이동하나요?

### 심화

- [ ] palette에 있는 color값들을 css 변수로 등록하고 사용했나요?
- [x]  비밀번호 input 요소 위에 비밀번호를 확인할 수 있는 눈 모양 아이콘을 추가했나요?

## 주요 변경사항

- Week1 위클리 미션이었던 index.html 파일을 수정하였습니다.
  - 불필요한 태그 삭제
  - &lt;main&gt; 을 &lt;article&gt; 로 변경
  - class 명 수정
  - &lt;main&gt;의 main-container 를 header 으로 클래스명 변경 후 &lt;header&gt; 로 이동
- html 파일의 class 명에 포함되던 "_"를 "-"으로 변경하였습니다.
- src 폴더에 css파일들을 옮겨주었습니다.

## 스크린샷

- CSS파일 정리
![CSS파일 정리](https://ifh.cc/g/0VnTac.png)

- index.html의 header 수정사항
![index.html의 header 수정사항](https://ifh.cc/g/GaW0vf.png)

## 멘토에게
- 배포된 사이트 주소입니다!
https://hilarious-souffle-d3c633.netlify.app/

- 템플릿 코드를 보고 난 후에 궁금한 점이 생겼습니다. 제가 느끼기에 템플릿 코드와 제가 작성했던 코드의 가장 큰 차이점은 "불필요하게 태그로 묶어주는 것"의 유무라고 생각했습니다. 저는 지금은 불필요하더라도 연관된 내용들을 하나의 태그로 묶어주면 나중에 제가 제 코드를 다시 볼때나 리팩토링 할 때에 읽고 작성하기 수월할 것이라고 생각했습니다. 하지만 템플릿 코드를 보니 불필요하게 묶지 않고 훨씬 직관적이었고 오히려 읽기도 쓰기도 훨씬 쉬울 것이라는 생각이 들었습니다. HTML 을 작성할 때에 나중을 생각해서 불필요한 태그를 만드는 것보다 최대한 직관적이고 단순하게 구현이 가능하다면 그 방법을 사용하고 필요한 부분은 그때그떄 수정하는 것이 바람직할까요?

Jnnsu committed:
feat: 기능 구현

Jnnsu committed:
feat: 기능 구현, week1 위클리 과제 수정

wonsss commented on FAQ.html:
```diff
@@ -0,0 +1,11 @@
+<body>
+  FAQ
+</body>
+</html>
```
파일의 마지막 줄에 개행을 하지 않으면, 이와 같이 표시됩니다.
EOL(End Of Line)을 넣어야 하는 이유와 운영체제별 EOL(EndOfLine) 차이로 인한 Git 문제 해결방법에 대해 알아보시면 좋을 것 같아요.

Jnnsu replied:
EOL에 대해서는 개념적으로 정리하였습니다. 운영체제별로 다른 EOL 때문에 문제가 발생할 수 있다는 것도 새롭게 알게되었습니다. 추가로 POSIX에 대해서도 블로그에 포스팅 해보겠습니다!

wonsss commented on index.html:
```diff
@@ -0,0 +1,103 @@
+        <a class="signup-button button" href="signup.html" target="_blank">링크 추가하기</a>
+        <img class="header_image" src="images/header-image.png" width="1118" alt="메인 이미지">
+      </div>
+    </header>
```
제 생각에는 로고와 로그인 버튼 부분은 header 요소에 해당될 수 있는데, " 세상의 모든 정보를 쉽게 저장하고 관리해보세요, 링크 추가하기" 부분은 main 요소로 이동되는 게 어떨까 싶어요. 준수님 생각은 어떠세요?

> HTML `<header>` 요소는 소개 및 탐색에 도움을 주는 콘텐츠를 나타냅니다. 제목, 로고, 검색 폼, 작성자 이름 등의 요소도 포함할 수 있습니다. 
https://developer.mozilla.org/ko/docs/Web/HTML/Element/header

> HTML `<main>` 요소는 문서 `<body>`의 주요 콘텐츠를 나타냅니다. 주요 콘텐츠 영역은 문서의 핵심 주제나 앱의 핵심 기능에 직접적으로 연결됐거나 확장하는 콘텐츠로 이루어집니다.
https://developer.mozilla.org/ko/docs/Web/HTML/Element/main

Jnnsu replied:
말씀해주신대로 main 요소에 들어가는 것이 바람직할 것 같습니다. 기존에 main 요소에 넣어두었다가 header로 옮겼었는데 너무 많은 생각이 오히려 독이 돼버렸네요..😂 

[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/08d167b062536bb7aadc76f986b3b1509e1c2955](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/08d167b062536bb7aadc76f986b3b1509e1c2955)

wonsss replied:
고민하면서 하셨다는 점이 너무 좋아요!

wonsss commented on index.html:
```diff
@@ -0,0 +1,103 @@
+        <img class="header_image" src="images/header-image.png" width="1118" alt="메인 이미지">
+      </div>
+    </header>
+    <article>
```
article도 괜찮은데, main 요소도 가능할 것 같아요!

Jnnsu replied:
이 부분도 기존에 main으로 했다가 article로 변경했는데 위와 비슷한 상황인 것 같습니다. 너무 템플릿 코드에 의존했던 것 같습니다. 마찬가지로 article에서 main으로 변경하였습니다!

[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/08d167b062536bb7aadc76f986b3b1509e1c2955](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/08d167b062536bb7aadc76f986b3b1509e1c2955)

wonsss commented on index.html:
```diff
@@ -0,0 +1,103 @@
+            </p>
+          </div>
+          <img src="images/landing/img1.png" width="550" height="450">
+      </section>
```
section 내에 제목 요소를 다 넣어주셨네요! 아래 MDN 명세처럼 잘 적용해주셨어요. 너무 좋아요~! 제목 요소를 잘 사용하는 것은 중요합니다. 1) 검색 엔진 최적화, 2) 웹 접근성, 3) 사용자 경험 등의 이유 때문이에요.

참고 아티클 [H1~H6 제목태그는 어떻게 사용해야하나요?](https://www.next-t.co.kr/seo/%ED%85%8C%ED%81%AC%EB%8B%88%EC%BB%ACseo/h1h6-%EC%A0%9C%EB%AA%A9%ED%83%9C%EA%B7%B8%EB%8A%94-%EC%96%B4%EB%96%BB%EA%B2%8C-%EC%82%AC%EC%9A%A9%ED%95%B4%EC%95%BC%ED%95%98%EB%82%98%EC%9A%94/) 간단히 읽어보세요!


> section은 각각의 `<section>`을 식별할 수단이 필요합니다. 주로 제목`h1~h6` 요소를 `<section>`의 자식으로 포함하는 방법을 사용합니다.
https://developer.mozilla.org/ko/docs/Web/HTML/Element/section

Jnnsu replied:
제목 요소에 대해서 어디에 어느 요소를 사용해야할지 감이 잘 안잡혔었는데 첨부해주신 링크를 확인해보니 궁금했던 점이 많이 해소되었습니다! 감사합니다.😄

wonsss commented on index.html:
```diff
@@ -0,0 +1,103 @@
+          <a href="https://www.facebook.com/" target="_blank"><img src="images/sns-icon/facebook-icon.svg" alt="페이스북 아이콘"></a>
+          <a href="https://twitter.com/" target="_blank"><img src="images/sns-icon/twitter-icon.svg" alt="트위터 아이콘"></a>
+          <a href="https://www.youtube.com/" target="_blank"><img src="images/sns-icon/youtube-icon.svg" alt="유튜브 아이콘"></a>
+          <a href="https://www.instagram.com/" target="_blank"><img src="images/sns-icon/instagram-icon.svg" alt="인스타그램 아이콘"></a>
```
img 요소에 alt 속성을 잘 적용해주셨네요! alt 속성도 웹 접근성 및 이미지 에러 시 대체 텍스트로서 중요합니다.

Jnnsu replied:
감사합니다! 작은 디테일들 놓치지 않도록 해보겠습니다.

wonsss commented on privacy-policy.html:
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
```
html 요소에 lang 속성을 en으로 주셨네요. 그러면 브라우저나 검색엔진 입장에서 이 html 문서는 영문용이라고 판단하게 됩니다. 

시각 장애인을 위한 스크린리더기가 이 언어를 인식하며 자동으로 음성을 변환하거나, 해당 언어에 적합한 발음을 제공합니다.
또한, 검색엔진은 검색 옵션에서 '한국어로 된 문서로' 범위를 제한할 경우 lang='ko' 인 문서를 우선으로 검색한다고 합니다.

아마 다른 파일은 ko로 다 바꾸셨던데, 이 부분은 별도로 개발을 시작하지 않으신 것 같아요. 참고로 알고 계시면 좋을 것 같아요

Jnnsu replied:
lang 속성을 en으로 준걸 미처 확인을 못했네요..!! 웹 접근성을 위해 꼭 신경써야 겠네요.

[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/37ae4d471a9691d99894a10878c33154cd0d3bc0](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/37ae4d471a9691d99894a10878c33154cd0d3bc0)

wonsss commented on privacy-policy.html:
```diff
@@ -0,0 +1,11 @@
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
검색해보니까, 이런 블로그 포스팅도 있더라구요. 이런 식으로 학습해보시는 것도 좋아요! 
[잘 모르고 넘어간 html 속성
](https://velog.io/@ansrjsdn/%EC%9E%98-%EB%AA%A8%EB%A5%B4%EA%B3%A0-%EB%84%98%EC%96%B4%EA%B0%84-html-%EC%86%8D%EC%84%B1)

Jnnsu replied:
링크해주신 글 읽어보았습니다. HTML 속성에 대해 다시 짚고 넘어갈 수 있었습니다!

wonsss commented on signin.html:
```diff
@@ -0,0 +1,42 @@
+    <section class="social-signin">
+      소셜 로그인
+      <div class="icons">
+        <a href="https://www.google.com/" target="_blank"><img src="images/social-icon/Component 2google-icon.png" alt="구글 아이콘" width="42" height="42"></a>
```
a 요소를 사용하실 때는, rel 속성에 noopener noreferrer  같은 값을 넣을지 말지 고민해보시는 것도 좋아요. 성능과 보안상 필요하실 수도 있습니다. 아래 링크 포스팅에 잘 설명되어 있으니 꼭 참고 바랍니다!

[하이퍼링크를 신뢰할 수 없다면? noopener, noreferrer, nofollow- 요즘 IT](https://yozm.wishket.com/magazine/detail/1586/)

[Link types -MDN](https://developer.mozilla.org/ko/docs/Web/HTML/Attributes/rel)

Jnnsu replied:
전혀 생각하지 못했던 부분인데 새로운 사실을 알게됐습니다. 해당 부분에 대해선 정리해서 블로그에 포스팅 해보도록 하겠습니다!

wonsss commented on signup.html:
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html lang="ko">
```
lang에 ko 값을 잘 적용해주셨네요 👍

Jnnsu replied:
감사합니다. 하지만 ko로 바꾸지 않은 부분들이 있었어서 조금 더 신경써야 할 것 같습니다.

wonsss commented on signup.html:
```diff
@@ -0,0 +1,47 @@
+      <div class="forms">
+        <label class="email form">
+          이메일<br>
+          <input name="signup-email" required>
```
input에 required 속성을 넣어주셨군요! 디테일 좋습니다!
그런데 input은 이미 여러 유즈케이스에 맞게 type들이 존재합니다.
email 용 input 이므로 `<input type="email"` 을 적용해주시면 더 좋을 것 같아요.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email

Jnnsu replied:
input type으로 email을 지정해주지 않는 실수를 해버렸네요..😱 해당 부분 수정완료하였습니다!

[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/7c5c2d230a78a8538e385cad2ed246a70d52c447](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/7c5c2d230a78a8538e385cad2ed246a70d52c447)

wonsss commented on signup.html:
```diff
@@ -0,0 +1,47 @@
+        </label>
+        <label class="password form">
+          비밀번호<br>
+          <input type="password" name="signup-password" required>
```
type="password" 좋습니다 👍

wonsss commented on signup.html:
```diff
@@ -0,0 +1,47 @@
+      <div class="forms">
+        <label class="email form">
+          이메일<br>
+          <input name="signup-email" required>
```
label과 input은 보통 연결되는 관련성을 지닐 수 있습니다.
label이 해당 input과 연결되기 위해서는 label에 for 속성을 넣어주고 input에 id를 넣어서 두 값을 같게 하면 됩니다

https://developer.mozilla.org/ko/docs/Web/HTML/Element/label
<img width="806" alt="image" src="https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/assets/59413128/90c3cf70-3ad1-433a-a607-55ef55e96cba">

Jnnsu replied:
label 내에 input을 넣어주었기에 id와 for로 연결시켜주지 않아도 된다고 생각했었습니다..! 이렇게 input이 label의 자식요소로 들어간 경우에도 for과 id를 넣어주는게 바람직하겠네요. 

[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/7b26111b5d249d7de4a3c3f60b40226a1132f7ec](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/7b26111b5d249d7de4a3c3f60b40226a1132f7ec)

wonsss commented on signin.html:
```diff
@@ -0,0 +1,42 @@
+        </h1>
+        <span>회원이 아니신가요? </span><a href="signup.html" class="signup-link">회원 가입하기</a>
+      </div>
+      <div class="forms">
+        <label class="email form">
+          이메일<br>
+          <input name="signin-email" required>
+        </label>
+        <label class="password form">
+          비밀번호<br>
+          <input type="password" name="signin-password" required>
+          <img class="password-switch-icon" src="images/password-icon/eye-on.svg" alt="비밀번호 표시 아이콘" width="16" height="16">
+        </label>
+      </div>
+      <button>로그인</button>
```
이렇게 정보를 제출하는 form의 경우 form 태그를 사용하시는 게 좋습니다.

<img width="782" alt="image" src="https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/assets/59413128/4e5c203e-7c58-4f8f-93bc-5e18a842602e">

https://developer.mozilla.org/ko/docs/Web/HTML/Element/form

Jnnsu replied:
form 태그로 수정하였습니다! 

[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/27f0939f92ff8107062b5c97828cc19f6e136d34](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/27f0939f92ff8107062b5c97828cc19f6e136d34)

wonsss commented on signin.html:
```diff
@@ -0,0 +1,42 @@
+          <img class="password-switch-icon" src="images/password-icon/eye-on.svg" alt="비밀번호 표시 아이콘" width="16" height="16">
+        </label>
+      </div>
+      <button>로그인</button>
```
submit button에 대해서도 알아보시면 좋아요.

https://developer.mozilla.org/ko/docs/Glossary/Submit_button

Jnnsu replied:
button의 기본값이 submit이라고 알고 있었기에 따로 type을 정해두지 않았었는데, type은 무조건 명시해두는 것이 좋은걸까요?
우선 type을 submit으로 지정해주었습니다.

 [https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/27f0939f92ff8107062b5c97828cc19f6e136d34](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/27f0939f92ff8107062b5c97828cc19f6e136d34)

wonsss replied:
명시적으로 하는 것이 좋습니다!

wonsss commented on signin.html:
```diff
@@ -0,0 +1,42 @@
+    <section class="signin">
+      <div class="signin-title">
+        <h1>
+          <a href="/"><img src="images/landing/logo.svg" alt="로고 이미지" width="210.583" height="38"></a>
```
링크나 이미지 모두 정상 동작하네요. 👍 
참고로 절대경로와 상대경로에 대해서 알아두시면 좋습니다.

[파일 경로 - 절대 경로와 상대 경로
](https://velog.io/@bami/%ED%8C%8C%EC%9D%BC-%EA%B2%BD%EB%A1%9C-%EC%A0%88%EB%8C%80-%EA%B2%BD%EB%A1%9C%EC%99%80-%EC%83%81%EB%8C%80-%EA%B2%BD%EB%A1%9C)

wonsss reviewed (commented):
안녕하세요 준수님. PR 내용을 깔끔히 잘 작성해주셨네요! 코드도 정성껏 작성하신 것 같아요. 👍 
배포된 주소를 PR 내용에 적어주셔서 결과물을 바로 확인 가능할 수 있어서 좋았습니다.

코멘트 단 내용들 확인해주시고 반영 가능한 부분들 반영하고 커밋 후 해당 커밋 링크와 함께 코멘트를 달아주시면 좋을 것 같습니다.

추가로 CSS 변수 사용과 eslint, prettier 설정도 해보시는 것을 제안드립니다. 그리고 prettier와 eslint 설정 후 커밋하시기 전에 `.gitignore` 파일에 대해 검색 후 이해하신 뒤에 커밋하시면 더 좋을 것 같습니다!

그리고 질문에 대한 답변입니다.

> 템플릿 코드를 보고 난 후에 궁금한 점이 생겼습니다. 제가 느끼기에 템플릿 코드와 제가 작성했던 코드의 가장 큰 차이점은 "불필요하게 태그로 묶어주는 것"의 유무라고 생각했습니다. 저는 지금은 불필요하더라도 연관된 내용들을 하나의 태그로 묶어주면 나중에 제가 제 코드를 다시 볼때나 리팩토링 할 때에 읽고 작성하기 수월할 것이라고 생각했습니다. 하지만 템플릿 코드를 보니 불필요하게 묶지 않고 훨씬 직관적이었고 오히려 읽기도 쓰기도 훨씬 쉬울 것이라는 생각이 들었습니다. HTML 을 작성할 때에 나중을 생각해서 불필요한 태그를 만드는 것보다 최대한 직관적이고 단순하게 구현이 가능하다면 그 방법을 사용하고 필요한 부분은 그때그떄 수정하는 것이 바람직할까요?

말씀하신 것처럼 직관적이고 단순하게 구현 가능하다면 그 방법을 사용하고 필요한 부분은 그때그때 수정하는 것도 좋습니다. 다만, 시맨틱 태그로서 넣은 부분들은 의미를 담기 위해 추가해주시면 좋습니다.
nav 내에 로고를 감싸던 h1 태그를 제거하시고, main 태그도 article 로 변경하셨던데, 제 생각으로는 기존 버전의 태그들이 시맨틱한 태그로서 역할을 하고 있었다고 볼 수 있습니다. 
 제 말이 정답이 아닐 수도 있어요. 상황에 따라 다를 수 있는 거니까요. 이와 같이 고민하시는 방향 너무 좋아요!

wonsss commented:
폴더 구조에 대해서도 한번 고민해보시면 좋을 것 같아요! 폴더 구조를 짜는 것도 프로그래밍 설계에서 중요한 부분입니다. 
폴더 구조에 정답은 없는 것 같아요. 다만 어떤 목적을 갖고 구조를 짜셨느냐가 중요합니다
응집성(Colocation) 이라는 키워드가 있습니다.  이를 중심으로 한번 생각해보시길 제안드립니다.

좋은 아티클이 있어 공유드립니다.
[프론트엔드에서 좋은 폴더 구조는 무엇일까?](https://xionwcfm.tistory.com/415)
[Javascript Colocation (번역글) 🤔](https://dkrnfls.tistory.com/408)

Jnnsu committed:
chore: eslint, prettier 추가

Jnnsu committed:
chore: gitignore 추가

Jnnsu committed:
refactor: <article>을 <main>으로 변경 후 header 블록을 <main>으로 이동

Jnnsu committed:
refactor: 시멘틱 태그 적용, class 네이밍 변경(BEM)

- 시멘틱 태그 적용: .header 클래스명.section-1 으로 변경, 기존의 section 블록을 article로 변경 후 section으로 감싸줌
- 전체적으로 class 네이밍 변경(BEM 규칙 적용)
- 시멘틱 태그 적용과 class 네이밍 변경 후 불필요한 CSS 제거

Jnnsu committed:
style: prettier의 tab 간격 4에서 2로 조정

Jnnsu committed:
chore: FAQ.html, privacy-policy.html의 lang=en을 lang=ko로 변경

Jnnsu committed:
chore: signin.html과 signup.html의 article을 main으로 변경

Jnnsu committed:
chore: email형식의 input type을 email로 지정

Jnnsu committed:
chore: input과 label을 id와 for로 연결

Jnnsu committed:
chore: div태그를 form태그로 변경, button type을 submit으로 지정

Jnnsu commented:
eslint와 prettier 설정 후 gitignore로 node_module 무시하도록 설정하였습니다!
[https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/dfbd16d42342d55520dfbf628510d8b6a1087850](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/dfbd16d42342d55520dfbf628510d8b6a1087850) [https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/5200256e54a1233c9d4d6a2b59a10a6043aa025f](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/5200256e54a1233c9d4d6a2b59a10a6043aa025f) [https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/d060fe99a47ed8b7928743efa747626ace6003a3](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/pull/31/commits/d060fe99a47ed8b7928743efa747626ace6003a3)

그리고 제가 질문드린 것에 대해 상세히 답변해주셔서 감사합니다. 답변을 달아주신게 정말 많은 도움이 됐습니다. 
단순히 구조를 짜는 상황만 보고 드린 질문이었지만, 어떻게 상황들을 풀어나가야 하는지에 대해 생각해 보게 됐습니다.
정답이라고 생각했던 것도 정답이 아닐 수 있고, 결국에는 상황에 따라 잘 판단해야 하는 것 같습니다. 
사실 피그마 시안을 보고 웹페이지를 만들어보는 과정에서 "최대한 명시된 것을 따르기"와 "유동적으로 알아서 잘 판단하기" 중에 어느 기준에 맞춰야할지 의문이었는데, 답변해주신 것과 이번 멘토링 때 조언해주신 것을 들으니 조금은 알 것 같습니다!! 감사합니다😃

Jnnsu committed:
chore: eslintrc.json과 prettierrc.json 수정

Jnnsu committed:
move: 폴더 구조 변경 및 파일 대폭이동, 일부 파일명 수정

Jnnsu committed:
chore: .gitignore에서 Icon 삭제

Jnnsu commented:
> 폴더 구조에 대해서도 한번 고민해보시면 좋을 것 같아요! 폴더 구조를 짜는 것도 프로그래밍 설계에서 중요한 부분입니다.
폴더 구조에 정답은 없는 것 같아요. 다만 어떤 목적을 갖고 구조를 짜셨느냐가 중요합니다
응집성(Colocation) 이라는 키워드가 있습니다. 이를 중심으로 한번 생각해보시길 제안드립니다.

어떻게 폴더를 응집시킬 것인가에 대해 알아보고 블로그에 정리해보았습니다! [velog](https://velog.io/@jspp120/%ED%8F%B4%EB%8D%94-%EA%B5%AC%EC%A1%B0)

아직은 완벽하게 이해되지 않는 부분(인덱싱, SSG, SSR)도 있었지만 어떤 목적을 가지고 폴더의 구조를 정할지에 따라 방법은 여러가지이고, 그 방법들을 통해 파일들을 응집시키는 것이 유지보수와 생산성을 높이기 위해 굉장히 중요하다는 것을 알았습니다. 앞으로 여러 프로젝트를 하면서 폴더 구조에 대해 꾸준히 고민해보도록 하겠습니다! 🧐

추가적으로 폴더 구조를 변경하는 과정에서 icon 폴더를 만들어주었는데, gitignore에 icon이 작성되어있던 것을 지우지 않고 push해 수정하였습니다.

Jnnsu committed:
docs: eslint 재설치, eslint/prettier ignore 생성, json 내용 수정

Jnnsu committed:
docs: .prettierrc.json 수정

Jnnsu committed:
move: 폴더 구조 변경 및 모든 파일 경로 수정

Jnnsu committed:
design: 로그인/회원가입 페이지 하단 배경색 적용 안되던 문제 수정

Jnnsu committed:
design: 메인 화면 header 중앙정렬 안되던 문제 수정

wonsss reviewed (approved):
준수님! 수고 많으셨습니다. 
리뷰를 꼼꼼하게 잘 반영해주셔서 감사해요.
블로그 글도 잘 읽었습니다. 정리를 정말 잘 하셨더라구요 👍 
Approve할게요! 다음 미션 코드리뷰에서 뵈어요.

Jnnsu committed:
refactor: 추가 브랜치 생성을 위해 push

Jnnsu committed:
refactor: 모바일 반응형 추가
